test/test-applications/regression/DuplicateTypeProxy/Program.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace DuplicateTypeProxy
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            await RunAsync(typeof(HttpClient));
+
+            for (int i = 0; i < 5; i++)
+            {
+                var assembly = Assembly.LoadFile(typeof(HttpClient).Assembly.Location);
+                await RunAsync(assembly.GetType("System.Net.Http.HttpClient"));
```

Comment:
A few iterations with Assembly.LoadFile is fine, but can you also use a custom ALC, to be sure we cover all the cases?

Reply:
done in 1e77200